pkg/operator/ceph/object/user.go

```diff
@@ -197,11 +198,17 @@ func setUserQuota(c *Context, id string, args []string) (string, int, error) {
 	args = append([]string{"quota", "set", "--uid", id}, args...)
 	result, err := runAdminCommand(c, args...)
 	if err != nil {
-		err = errors.Wrap(err, "failed to set max buckets for user")
+		if strings.Contains(result, "bucket name or uid is required for quota operation") {
```

Comment:
Why do we need special error codes here? This pattern of looking for error strings is less than ideal. If the string changes in a future version of ceph, this checking would be broken.

Reply:
I would even ask if we can change the method signature to return `(string, error)` instead of returning the error code.

Reply:
@travisn that make more sense, before this apis was always returning `RGWErrorNone`, consumer of this function checking that value for error check. So error returned from this api never check properly. So I changed format bit. I will get rid of `int` from this function. IMO same should applicable for all the apis in this file(may be in different PR)?